Dockerfile

```diff
@@ -13,5 +13,8 @@ FROM openjdk:11-jdk-slim
 ARG path=/usr/app
 WORKDIR ${path}
 COPY --from=build ${path}/target/*.jar ${path}/app.jar
+
 EXPOSE 8080
-ENTRYPOINT ["java","-jar","app.jar"]
\ No newline at end of file
+
+#ENTRYPOINT ["java","-jar","app.jar", "-Dserver.port=$PORT", "-jar target/*.jar"]
+CMD [ "sh", "-c", "java -Dserver.port=$PORT -Xmx280m -Xss512k -XX:CICompilerCount=2 -Dfile.encoding=UTF-8 -XX:+UseContainerSupport -Djava.security.egd=file:/dev/./urandom -jar app.jar"]
```

Comment:
I'm interested in several options:
-Xss512k - why do you need to limit stack size?
-XX:CICompilerCount=2
[Article:](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/java.html)
> Sets the number of compiler threads to use for compilation. By default, the number of threads is set to 2 for the server JVM, to 1 for the client JVM, and it scales to the number of cores if tiered compilation is used. The following example shows how to set the number of threads to 2:

-Djava.security.egd=file:/dev/./urandom
[Article:](https://www.baeldung.com/java-security-egd#what-is-javasecurityegd)
> Typically, if we're running Java 8 or later and are running on Linux, then our JVM will use file:/dev/urandom by default.

 Which goals are they added for?

Reply:
Why did you switch from ENTRYPOINT to CMD?

Reply:
i removed it, because -XX:CICompilerCount=2 and -Xss512k are default values for my type Dyno container

-Djava.security.egd=file:/dev/./urandom - I read about this , it s because: https://qastack.ru/programming/58991966/what-java-security-egd-option-is-for

Reply:
Why did you switch from ENTRYPOINT to CMD?

as I understood there is not much difference, since I saw most of the examples using cmd, I changed this